### PR TITLE
Fix email lookup in invite dialog

### PR DESCRIPTION
**Pull request by weeman1337**
Before, the invitation dialog filtered the recent and removed duplicates in all results including suggestions.
Now, it filters by the search term and then removes duplicates.

| Before | After |
| --- | --- |
| image | image |

closes https://github.com/vector-im/element-web/issues/23353

PSF-1898

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Fix email lookup in invite dialog ([\#10150](https://github.com/matrix-org/matrix-react-sdk/pull/10150)). Fixes vector-im/element-web#23353.

### CODE REVIEW

File: src/components/views/dialogs/InviteDialog.tsx [hunk 1]
```diff
@@ -684,7 +684,8 @@ export default class InviteDialog extends React.PureComponent<Props, IInviteDial
                                 display_name: profile.displayname,
                                 avatar_url: profile.avatar_url,
                             }),
-                            userId: lookup.mxid,
+                            // Use the search term as identifier, so that it shows up in suggestions.
+                            userId: term,
```
Comment (t3chguy) [inline src/components/views/dialogs/InviteDialog.tsx:688]: This is brittle and confusing, this isn't a userId (read MXID). In all other cases `userId` maps to `user.X` where X is the key which holds the MXID. Feels like you should add a `filter_term` field to `threepidResultsMixin` results which can be used to hold the original query or email or whatever you like more generically. Or maybe a `is_match=true` or similar.

**Review by t3chguy — CHANGES_REQUESTED** [review]

File: src/components/views/dialogs/InviteDialog.tsx (hunk 1 above)
Comment (weeman1337) [reply src/components/views/dialogs/InviteDialog.tsx:688]: It is already used this way in other places  :wink:  https://github.com/matrix-org/matrix-react-sdk/blob/develop/src/components/views/dialogs/InviteDialog.tsx#L654-L657

I can try to change it anyway.

Comment (t3chguy) [reply src/components/views/dialogs/InviteDialog.tsx]: Ah yes, did miss that incantation. My bad.

Comment (t3chguy) [reply src/components/views/dialogs/InviteDialog.tsx]: I think I prefer your simpler first solution :P

**Review by t3chguy — APPROVED** [review]
:shipit:

**Pull request merged**